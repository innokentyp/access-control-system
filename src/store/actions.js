import { 
	LOGIN,

  LOGIN_SUCCEEDED, 
  LOGIN_FAILED,

  LOGOUT
} from './constants'

export function login(credentials) {
	return {
		type: LOGIN,
		credentials
	}
}

export function loginSucceeded(user) {
	return {
		type: LOGIN_SUCCEEDED,
		user
	}
}

export function loginFailed(message) {
	return {
		type: LOGIN_FAILED,
		message
	}
}

export function logout() {
	return {
		type: LOGOUT
	}
}