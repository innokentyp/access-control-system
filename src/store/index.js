import { createStore, applyMiddleware } from 'redux'

import thunkMiddleware from 'redux-thunk'
import createSagaMiddleware from 'redux-saga'

import { createLogger } from 'redux-logger'

import reducer, { preloadedState } from './reducers'
import requestSubjectsSaga from './sagas/requestSubjectsSaga'

export default function configureStore() {
	// create the saga middleware
	const sagaMiddleware = createSagaMiddleware()

	const store = createStore(reducer, preloadedState, applyMiddleware(thunkMiddleware, sagaMiddleware, createLogger()))  

	// then run the saga
	sagaMiddleware.run(requestSubjectsSaga)

	// return
	return store
}