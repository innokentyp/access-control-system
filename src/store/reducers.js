import { combineReducers } from 'redux'

import * as types from './constants'

// constants
personal_editor.MAX_SUBJECTS = 12

// preloaded state
export const preloadedState = {
  authentication: {
    user: {}
  },

  personal: {
    subjects: [],

    filtering: '',
    sorting: ''
  },

  personal_editor: {
    subjects: {}
  } 
}

// reducers
function authentication(state = preloadedState.authentication, action) {
  switch (action.type) {
    case types.LOGGED_IN:
      return { ...state, user: { ...action.user } }
    case types.LOGGED_OUT:
      return { ...state, user: {} }  
    default:
      return state 
  }
}

function personal(state = preloadedState.personal, action) {
  switch (action.type) {    
    case types.SUBJECTS_FETCHED:
      return { ...state, subjects: action.subjects }

    case types.SET_FILTERING: 
      return { ...state, filtering: action.filtering }  
    case types.SET_SORTING: 
      return { ...state, sorting: action.sorting }  
    default:
      return state 
  }
}

function personal_editor(state = preloadedState.personal_editor, action) {
  switch (action.type) {    
    case types.SUBJECT_FETCHED:
      const subjects = { ...state.subjects, [action.subject.id]: action.subject }
      const ids = Object.keys(subjects)

      while (ids.length > personal_editor.MAX_SUBJECTS) {
        delete subjects[ids[0]]
        ids.splice(0, 1)
      }

      return { ...state, subjects }    
    default:
      return state 
  }
}

export default combineReducers({ authentication, personal, personal_editor })