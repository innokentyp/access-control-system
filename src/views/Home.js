import React, { Component } from 'react'
import { Container } from 'semantic-ui-react'

class Home extends Component {
	render() {
		return (
			<Container as="section">
				<h3>{this.constructor.name} match <code>{this.props.match.url}</code> for <code>{this.props.location.pathname}</code></h3>

				<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Rerum commodi non, laudantium quia est, recusandae debitis mollitia adipisci, ea, sapiente expedita. Dignissimos cumque quisquam, voluptate, sit obcaecati sed! Ut fugit quas a assumenda magnam, nesciunt minima fuga expedita obcaecati voluptas suscipit. Veniam ut molestiae unde reprehenderit quia minus, earum voluptates rerum error eos veritatis aliquam consectetur cum accusamus ullam, a incidunt. Suscipit est, dolores voluptatem minima temporibus eaque voluptatibus unde cum animi quidem cumque adipisci repudiandae repellendus, ullam accusantium quos eum reiciendis quibusdam, eos aliquam! Culpa pariatur, animi iure omnis beatae repellendus praesentium totam sunt enim distinctio. Eaque quibusdam molestias dolorum exercitationem officia sed repudiandae ullam sunt beatae fuga. Veniam corrupti accusamus magni quasi, facere officiis accusantium rerum, aperiam quam eos autem nostrum vero inventore ipsum exercitationem dolorem unde nulla, sapiente. Sequi, tempora, voluptate, veniam laboriosam quaerat cupiditate temporibus, repudiandae alias natus asperiores magnam quia labore possimus eligendi iste. Fugiat consequuntur nemo nobis assumenda dolorum consectetur quae, deleniti, beatae totam obcaecati voluptatum. Placeat, nihil, atque ex dolore, iste expedita, non voluptas et itaque cumque ratione dolorem. Modi dolor non fuga ab nihil aperiam blanditiis quod recusandae quos animi officia, dicta sapiente in aut a dolorum vitae voluptatibus cupiditate deserunt. Et officia ullam, sint quaerat voluptas deserunt explicabo eum itaque laudantium, quas quidem minus consequatur vel? Aliquam deserunt pariatur repellat, aut explicabo tenetur, id obcaecati ipsa dolor. Laudantium architecto consequuntur suscipit expedita, quaerat maxime ipsum! Ut officia eaque doloremque sint quae assumenda iusto, culpa dicta earum. Esse earum hic placeat eveniet ipsa. Neque tenetur dolore ipsa atque sit laborum, perferendis fugit, odio. Exercitationem modi ea ullam officiis veniam mollitia possimus quam quis fugiat soluta, ipsam est consequuntur voluptatibus quas nostrum, fugit, perferendis molestias? Ea dolorem cum nisi, id numquam dolores illo natus impedit nostrum recusandae velit tenetur facere, explicabo, ducimus omnis expedita. Sunt amet delectus commodi libero ratione vitae aspernatur numquam ducimus aliquam et quia, illo eveniet fugiat facere voluptates provident! Tempore enim tenetur magnam odio aperiam eius, vero quis minus repudiandae. Veniam architecto, atque quam reiciendis cum, nobis ducimus distinctio. Illum voluptatibus quasi animi temporibus non impedit unde explicabo deserunt nisi cumque? Corporis ad eligendi sapiente amet aliquam perspiciatis repudiandae tempore sunt impedit, iusto, facilis, odit fugit, molestias quas. Totam, repudiandae vel minima quo ratione soluta porro eaque a, provident non quisquam quam, est consectetur rerum aliquid et in officiis, blanditiis nobis! Veniam eos cumque iure! Porro autem deleniti, eligendi esse quaerat accusamus soluta possimus rem, incidunt odio. Minus qui optio explicabo est similique facilis, blanditiis, nam ipsam illo nesciunt sint. Recusandae obcaecati, repudiandae. Deserunt dignissimos harum assumenda ut quas sequi enim, architecto illo repudiandae. Explicabo voluptatem perspiciatis aperiam corrupti, laborum! Deleniti libero delectus necessitatibus optio molestias dignissimos corrupti assumenda repellendus quasi autem, quo officia placeat corporis vitae sunt nobis quos consequuntur. Necessitatibus, quod quo aperiam voluptatem. Praesentium voluptates eius natus? Praesentium, voluptatem. Blanditiis fugit aspernatur doloremque expedita nisi quasi pariatur mollitia alias illum modi, voluptas suscipit est esse laudantium impedit incidunt officiis repellendus dolorem id, odit! Autem rem repellat sequi, earum accusantium esse nihil et molestiae doloribus quos iure, ducimus officia veniam odio atque sunt omnis maxime, inventore minus asperiores. Nobis rem, dicta nemo, porro dolorem quisquam laboriosam minus similique consequatur dolor quo esse officia voluptatem temporibus, a obcaecati adipisci velit assumenda culpa. Architecto, repudiandae iure enim. Ratione quae impedit quia, non amet soluta illum repudiandae ad unde minus deleniti dolorem consequuntur eius, qui consequatur in a, asperiores. Deserunt culpa id veritatis at ratione voluptate, facilis debitis hic optio rerum repudiandae quidem non fugit architecto? Delectus quam eaque impedit, officia aspernatur fugit autem doloremque eum perspiciatis? Dignissimos aperiam nesciunt neque ab quas eligendi autem, recusandae, quae aspernatur, quasi consequatur quia reprehenderit fugit doloribus enim quo vitae totam aut rem unde voluptatibus! Aspernatur fugit deserunt repellendus architecto ipsa ducimus illo beatae modi dignissimos optio, est quia molestiae, amet non at nam reprehenderit? Ipsam assumenda repudiandae officiis impedit ducimus perferendis quam rem labore, minus voluptate totam maxime repellendus quos reiciendis neque, quae consequatur itaque. Culpa eveniet deleniti assumenda fugiat ex, eligendi molestias doloribus animi consectetur possimus hic doloremque laboriosam dolore, natus harum obcaecati et amet perferendis sit. Voluptatum praesentium deleniti illum, aspernatur natus ducimus voluptate laborum repudiandae, odio quae obcaecati iusto nostrum accusamus doloremque, sint exercitationem fugiat, nesciunt pariatur dicta fuga aperiam. Harum ea recusandae dolores consectetur. Quod dolorum est, expedita temporibus vero tempore blanditiis possimus maiores nulla? Possimus repudiandae, aperiam, officia veniam rerum assumenda numquam vitae voluptatum aliquid. Ducimus id velit nobis blanditiis, doloremque, eius ad neque facere sequi consequuntur eligendi at rem. Ex voluptates nobis praesentium porro magnam optio veritatis similique quibusdam velit incidunt ab maiores suscipit, obcaecati, nostrum adipisci nemo natus numquam fugiat eveniet perspiciatis ipsa earum animi. Atque temporibus voluptate vero impedit ullam, molestiae quaerat, iusto laudantium sint quisquam fugit, possimus illum. Assumenda tempore porro ad ullam commodi temporibus ipsum vero accusamus magnam officiis cumque, cum reprehenderit illo expedita sunt libero, saepe optio omnis sapiente modi delectus, facere eum quibusdam ipsa repellendus. Accusantium corporis doloribus maiores cupiditate sunt, deleniti perferendis quas earum beatae quia nisi, ab fugiat qui dolore consectetur architecto. Veniam ab corrupti ullam nulla rem iusto beatae, modi quos natus, distinctio itaque, officia dolorum, laudantium? Odio unde quos quaerat laborum, in, veniam dolorum vel blanditiis distinctio id ipsum amet deleniti nulla adipisci neque harum, iure dignissimos. Impedit, dolore, cum. Id voluptate illum et error vitae suscipit harum magnam sunt dicta quibusdam repellendus, voluptas porro maxime laboriosam eaque in laborum provident deserunt quisquam ipsum nostrum. Dolores consequatur excepturi nesciunt animi reiciendis perferendis porro temporibus, dolore dolorum sit similique debitis. Omnis dicta consequatur quisquam, tempore accusamus excepturi magni autem quidem atque voluptatum ipsam neque similique doloribus esse ipsa quam vitae dolore mollitia temporibus, impedit voluptatibus. Suscipit, quasi, itaque. Nobis voluptatibus error dolores commodi fugiat, odit doloremque deleniti eos sunt pariatur, aperiam atque distinctio similique optio minus debitis voluptate sapiente? Expedita accusantium quis repudiandae nobis voluptates consequatur, labore magni. Omnis in inventore corporis rerum aliquam officiis hic sit, dignissimos impedit saepe voluptas, quibusdam nesciunt consectetur natus? Neque earum recusandae eveniet ipsa vel ratione omnis quo ut. A blanditiis eveniet debitis amet vero, itaque numquam hic magni voluptatibus libero laborum, mollitia quo aperiam.</p>	
			</Container>
		)
	} 
}

export default Home