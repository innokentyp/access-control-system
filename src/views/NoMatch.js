import React, { Component } from 'react'
import { Container } from 'semantic-ui-react'

export default class NoMatch extends Component {
	render() {
		return (
			<Container as="section">
		    <h3>No match for <code>{this.props.location.pathname}</code></h3>
		  </Container>
		)
	} 
}

