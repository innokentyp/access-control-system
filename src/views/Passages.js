import React, { Component } from 'react'
import { Container } from 'semantic-ui-react'

class Passages extends Component {
	render() {
		return (
			<Container as="section">
				<h3>{this.constructor.name} match <code>{this.props.match.url}</code> for <code>{this.props.location.pathname}</code></h3>	
			</Container>
		)
	} 
}

export default Passages