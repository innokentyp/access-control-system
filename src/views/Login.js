import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { Container, Form, Button, Message } from 'semantic-ui-react'

import * as actions from '../store/actions/authentication'

class Login extends Component {
	state = {
		loading: false,
		warning: {},
		error: {}	
	}

	constructor(props) {
    super(props)

    this.inputUser = React.createRef()
    this.inputPassword = React.createRef()
  }

  validate = (e) => {
  	var warning = {}
		var validated = true

		if (!e.target.user.value) {
			warning.user = 'Имя пользователя не может быть пустым'
			
			validated = false
			this.inputUser.current.focus()
		}

		if (!e.target.password.value) {
			warning.password = 'Пароль не может быть пустым'
			
			if (validated) {
				validated = false
				this.inputPassword.current.focus()
			}
		}

		this.setState({ warning })

		return validated
  }

	formLoginSubmit = (e) => {
		e.preventDefault()

		this.setState({ error: {} })

		if (this.validate(e)) {
			this.setState({ loading: true })

			this.props.actions.login({ name: e.target.user.value, password: e.target.password.value, remember: e.target.remember.checked })
				.then (
					() => {
						let { history, location } = this.props

						location.state ? history.push(location.state.from.pathname === location.pathname ? '/' : location.state.from.pathname) : history.push('/')
					}
				)
				.catch (
				  (error) => {
				  	this.setState({ loading: false, error })

				  	this.inputUser.current.select()				    
				  }
			  )
			  .finally (
					() => {
						;
					}
				)				
		} 
	}

	formLoginReset = (e) => {
		this.setState({ warning: {}, error: {} })

		this.inputUser.current.focus()
	}

	render() {
		var { loading, warning, error } = this.state

		return (
			<Container as="section" text>
				<h3>Login match <code>{this.props.match.url}</code> for <code>{this.props.location.pathname}</code></h3>
				
				<Form name="form-login" onSubmit={this.formLoginSubmit} onReset={this.formLoginReset} autoComplete="off" loading={loading} error={!!Object.keys(error).length} warning={!!Object.keys(warning).length}>		
		      <Form.Field error={!!warning['user']}>
		      	<label htmlFor="form-login-user">Имя пользователя:</label>
			      <input ref={this.inputUser} type="text" name="user" id="form-login-user" autoComplete="nope" autoCorrect="off" autoCapitalize="off" spellCheck="false" autoFocus />
		      </Form.Field>

			    <Form.Field error={!!warning['password']}>
			      <label htmlFor="form-login-password">Пароль:</label>
			      <input ref={this.inputPassword} type="password" name="password" id="form-login-password" autoComplete="new-password" />
			    </Form.Field>

		      <Form.Checkbox name="remember" label="Запомнить меня" />

		      <Message
			      warning
			      header="Ошибки при вводе данных"
			      list={Object.values(warning)}
			    />
			
					<Message
			      error
			      header="Неверное имя пользователя или пароль"
			      content={error.message ? error.message : 'Нет сообщения'}
			    />
					
			    <Button type="submit" primary>Login</Button>
			    <Button type="reset">Reset</Button>			    
			  </Form>
			</Container>
		)
	} 
}

export default connect(
	state => (
		{ 
			authenticated: !!state.authentication.user.jwt
		}
	), 
	dispatch => (
		{ 
			actions: bindActionCreators(actions, dispatch)			
		}
	)
)(Login)