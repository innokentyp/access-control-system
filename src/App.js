import React, { Component, Fragment } from 'react'
import { Route, Switch } from 'react-router-dom'

//import logo from './logo.svg'
//import './App.css'

import Navigation from './components/Navigation'
import PrivateRoute from './routes/PrivateRoute'

import Home from './views/Home'
import Passages from './views/Passages'
import Archive from './views/Archive'
import Personal from './views/Personal'

import Login from './views/Login'
import NoMatch from './views/NoMatch'

class App extends Component {
  render() {
    return (
      <Fragment>
        <Navigation />

        <Switch>
          <Route path='/' exact component={Home} />
          
          <PrivateRoute path='/passages' component={Passages} />
          <PrivateRoute path='/archive' component={Archive} />
          <PrivateRoute path='/personal' component={Personal} />

          <Route path='/login' exact component={Login} />
          <Route component={NoMatch} />
        </Switch>
      </Fragment>
    )
  }
}

/*
class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    )
  }
}
*/

export default App
