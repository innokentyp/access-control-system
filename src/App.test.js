import React from 'react'
//import { render, unmountComponentAtNode } from 'react-dom'
//import ReactTestUtils from 'react-dom/test-utils'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router'

import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

//import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

import configureMockStore from 'redux-mock-store'

import App from './App'
import Home from './views/Home'
import Login from './views/Login'
import Passages from './views/Passages'
import NoMatch from './views/NoMatch'

import /*reducer, */{ preloadedState } from './store/reducers'

const middlewares = []
const mockStore = configureMockStore(middlewares)

configure({ adapter: new Adapter() })

describe('App', 
	() => {
		it('renders without crashing', 
			() => {
			  shallow(<App />)			  
			}
		)

		it('navigate to public route without logged in', 
			() => {
				const wrapper = mount(
					<Provider store={mockStore(preloadedState)}>
						<MemoryRouter initialEntries={[ '/' ]}>
							<App />
						</MemoryRouter>
					</Provider>
				)

				expect(wrapper.find(Home)).toHaveLength(1)
			}
		)

		it('show login if navigate to private route', 
			() => {
				const wrapper = mount(
					<Provider store={mockStore(preloadedState)}>
						<MemoryRouter initialEntries={[ '/passages' ]}>
							<App />
						</MemoryRouter>
					</Provider>
				)

				expect(wrapper.find(Login)).toHaveLength(1)
			}
		)

		it('navigate to private route if logged in', 
			() => {
				const state = { 
					...JSON.parse(JSON.stringify(preloadedState)), 
					authentication: {
				    user: {
				    	name: 'user', 
				    	jwt: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoidXNlciIsInBhc3N3b3JkIjoiMTIzNDU2In0.F1kpu314aOcXQjugsxi8XZzV4dWqE7qyOMcxzJN7zTI'
				    }
				  }
				}

				const wrapper = mount(
					<Provider store={mockStore(state)}>
						<MemoryRouter initialEntries={[ '/passages' ]}>
							<App />
						</MemoryRouter>
					</Provider>
				)

				expect(wrapper.find(Passages)).toHaveLength(1)				
			}
		)

		it('navigate to no match if route no match', 
			() => {
				const wrapper = mount(
					<Provider store={mockStore(preloadedState)}>
						<MemoryRouter initialEntries={[ '/qwerty' ]}>
							<App />
						</MemoryRouter>
					</Provider>
				)

				expect(wrapper.find(NoMatch)).toHaveLength(1)
			}
		)

		it('navigate on click after logged in', 
			() => {
				const state = { 
					...JSON.parse(JSON.stringify(preloadedState)), 
					authentication: {
				    user: {
				    	name: 'user', 
				    	jwt: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoidXNlciIsInBhc3N3b3JkIjoiMTIzNDU2In0.F1kpu314aOcXQjugsxi8XZzV4dWqE7qyOMcxzJN7zTI'
				    }
				  }
				}
				
				const Assert = withRouter(
					class _Assert extends React.Component {
						componentDidMount() {
							const { location: { pathname }, history } = this.props

							expect(pathname).toMatch(new RegExp('/$'))

							history.push('/passages')
						}

						componentDidUpdate() {
							const { location: { pathname } } = this.props

							expect(pathname).toMatch(new RegExp('/passages$'))
						}

						render = () => <App />						
					}
				) 

				const wrapper = mount(
					<Provider store={mockStore(state)}>
						<MemoryRouter initialEntries={[ '/' ]}>
							<Assert />
						</MemoryRouter>
					</Provider>
				)

				expect(wrapper.find('Link[to="/passages"]')).toHaveLength(1)
				expect(wrapper.find(Passages)).toHaveLength(1)

				/*
				const div = document.createElement('div')

				class _Assert extends React.Component {
					componentDidMount() {
						const { location: { pathname }, history } = this.props

						console.log(`1: ${pathname}`)

						const link = div.querySelector('[href="/passages"]')
						expect(link).not.toBeNull()

						//ReactTestUtils.Simulate.click(link)
						history.push('/passages')
					}

					componentDidUpdate() {
						const { location: { pathname } } = this.props

						console.log(`2: ${pathname}`)

						expect(pathname).toMatch(new RegExp('/passages$'))
					}

					render = () => <App />						
				}

				const Assert = withRouter(_Assert) 

				render(
					<Provider store={createStore(reducer, state)}>
						<MemoryRouter initialEntries={[ '/' ]}>
							<Assert />
						</MemoryRouter>
					</Provider>, 
					div
				)

				unmountComponentAtNode(div)
				*/
			}
		)		

	}
)	