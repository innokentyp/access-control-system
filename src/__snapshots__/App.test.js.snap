// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<React.Fragment>
  <withRouter(Connect(Navigation)) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/passages"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/archive"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/structure"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      exact={true}
      path="/personal"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/personal/new"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/personal/:id"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/access"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/accounting"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/security"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</React.Fragment>
`;
